### DISCUSSION TITLE
Feature request: "Builds of the Month" page

### DISCUSSION THREAD

**Discussion by LuminairPrime** [Heroes-Profile/heroesprofile#696] [Ideas]
I just posted this on Reddit in response to the news that you're rebuilding HeroesProfile from scratch. The site was always incredible compared to the alternatives. I hope you're having a lot of fun working with the new version!!

I have a feature request: a "**Builds of the Month**" page!

I don't play enough to remember the best talent builds, so every time I do play I visit HeroesProfile to look things up. But that's slow and costs you CPU time. All I really need is a matrix of the best build codes: heroes on the Y axis, and game modes on the X axis. Obviously 1 top build doesn't tell you EVERYTHING; it's not the best build all the time, but for me as a casual, it's the start I need.

Since the game doesn't change enough to warrant loading a fresh statistic, you can build this list just once a month. A single list will also make it easy to sit down for half an hour and populate the build save slots of every hero in the game: slot 1 for storm league, slot 2 for quick match, and slot 3 for ARAM.

I can imagine enhancements to this feature. Old BOTM pages could be archived, and new cells with new builds that month could be highlighted in a different color. You could calculate the "rising build of the month": whichever build had the biggest change in popularity or winrate compared to last month. Each build text could even have a mouseover showing the winrate of that build and its best map, giving the reader extra context.

I thought of this feature the other day because I wanted to play ARAM and I needed to load a LOT of HeroesProfile build pages. With Builds of the Month I could load just one page!

Keep up the good work, your site is amazing and keeps the game alive. I wouldn't remember how to play without your help LOL!

Edit 1:

I've created a mockup below. When I'm looking at winrates in my free time, I pick the largest sample size possible for more accuracy. I haven't seen much benefit in trimming down the dataset by player rank or whatever. I just pick the largest range of patches possible. I also often combine Quick Match and Storm League data when looking for the best build, but in the case of BOTM, the builds of the game modes are next to each other, so you can eyeball if the difference matters. 

For this mockup I selected "Major Patch", but maybe "Last Update" is the way to go as long as there's over 30 days of data for it? 

Possible enhancement: In the event of a game patch right before a BOTM is generated, the affected heroes could have their last top build shown with a different color highlighting that it was valid in the previous patch and there isn't enough data for an update to it yet.

The devil is in the details when it comes to choosing the #1 build for a hero. I'm looking at Ana in Storm League: the #1 build might only have a small fraction of the player data as the #2 build, and the #1 might only have a fractional improvement in winrate over #2 as well, so maybe the #2 build would be the correct one to recommend. Or maybe the existing HeroesProfile algorithm would be best., I'm not sure. I wouldn't let perfection be the enemy of the good here, though; I would be fine playing the game with some statistical artifact builds.

Funny possible enhancement: if a hero has the same build for all 3 game modes, insert a golden C-C-COMBO after the first build field 🤣

Mockup in Excel:

![image](https://github.com/Heroes-Profile/heroesprofile/assets/91862358/2ed82487-d2e1-4e1f-94a0-91af9f140fcb)

**Comment by Zemill** [comment]
I like it.  Let me mull it over and see how it might be fit

**Comment by Zemill** [comment]
This has been added to some affect on https://www.heroesprofile.com/Global/Talents

<img width="1737" height="1039" alt="image" src="https://github.com/user-attachments/assets/09bf8186-8c66-4285-ba8a-a974d4eb2ba9" />


This is likely the extent that we can make towards this, as any other approach will require some daily task on the backend to calculate
